src/app.ts

```diff
@@ -153,8 +162,56 @@ export type ContextData = {
 
 const fedifyKv = await KnexKvStore.create(client, 'key_value');
 
+let messageQueue: MessageQueue | undefined;
+
+if (process.env.USE_MQ === 'true') {
+    logging.info('Message queue is enabled');
+
+    try {
```

Comment:
Everything in this block feels like "factory" code to me - I think it would be a lot nicer to have a static create factory method on the class, which can handle the assertions that the topic exists etc...

```
class GCloudPubSubMessageQueue {  
  ...
  async static create() {
    // all the init checks etc...
    return new GCloudPubSubMessageQueue(args);
  }
}
```

You can then mark the `constructor` as `private` and be sure that any instance of the class has had all of the requirements checked and met

Reply:
Yeh, i did actually start down a similar path to this, but outside of the class itself (felt better, made easier to test). See what you think → https://github.com/TryGhost/ActivityPub/pull/143/commits/ea15558c1e29fc888198c391a5f855156ed2ed29